### PR TITLE
gstreamer: add linbice dependency

### PR DESCRIPTION
**Pull request by aconchillo**
This enables the nice plugins and also adds the webrtcbin plugins from gst-plugins-bad.

Note, that this is different from the disabled libnice subproject. The subproject clones libnice and try to install it in the system, which we don't want. In contrast, adding the dependency just allows plugins to depend on the system installed one.

- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

### CODE REVIEW

**Comment by aconchillo** [comment]
The caveats instruction is necessary because that's actually the real GStreamer plugin path. Otherwise, GStreamer will load plugins from the Cellar path instead and, for example, it won't load the gstreamer plugin that comes in `libnice`.

**Comment by aconchillo** [comment]
@chenrui333 any chance to add the `Ci-long-timeout` label again? Let me know if there's anything missing.

Btw, I'm also working on a current issue. Basically lots of GStreamer plugins are broken (at least on Apple Silicon), try for example:

```
gst-launch-1.0 videotestsrc ! fakesink
```

This is now fixed here: https://gitlab.freedesktop.org/gstreamer/orc/-/merge_requests/163. Once this is merged, I believe they will release a new ORC version. After that we will need a new revision of GStreamer to compile orc files with the new ORC.

**Review by carlocab — CHANGES_REQUESTED** [review]
`libnice` depends on `gstreamer`, so this creates a circular dependency (which we can't have).

Also, setting `GST_PLUGIN_SYSTEM_PATH` seems to be discouraged by upstream maintainers. See #120206.

**Comment by aconchillo** [comment]
> `libnice` depends on `gstreamer`, so this creates a circular dependency (which we can't have).
> 
> Also, setting `GST_PLUGIN_SYSTEM_PATH` seems to be discouraged by upstream maintainers. See #120206.

Should we create a `gstreamer-libnice` package that depends on `gstreamer` and then `gstreamer` can safely depend on `libnice`?

**Comment by aconchillo** [comment]
> > `libnice` depends on `gstreamer`, so this creates a circular dependency (which we can't have).
> > Also, setting `GST_PLUGIN_SYSTEM_PATH` seems to be discouraged by upstream maintainers. See #120206.
> 
> Should we create a `gstreamer-libnice` package that depends on `gstreamer` and then `gstreamer` can safely depend on `libnice`?

I'll give that a try...

**Comment by aconchillo** [comment]
Closing for now until I have something working locally.

**Pull request closed without merging**
